### DISCUSSION TITLE
Disable insecure legacy MFA methods

### DISCUSSION THREAD

**Discussion by k12-sysadmin** [community/community#133944] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have no use for legacy, insecure methods of authentication such as TOTP codes or SMS text messages, but GitHub will not allow me to disable both of those methods.  It seems to require one of these weak methods always be enabled.

I primarily log into GitHub's web GUI with a Passkey (which is stored on a hardware security key).  I use git command-line tools exclusively with an SSH key (which is also stored on a hardware security key).

My main security key is always with me, and my recovery code and a second security key are safely stored in a physically secure location.  I know how to manage an account with strong authentication and not get myself locked out; I have had my Google account enrolled in the Advanced Protection Program for quite some time.  I don't need my level of security dumbed down for me.  I assume most people using a platform like GitHub are in the same boat with me in that regard.

GitHub, please let us remove legacy authenticators from our accounts!

**Comment by Sivamani-18** [comment]
I understand your frustration with GitHub's security policies. It seems that despite your preference for more robust authentication methods like passkeys and hardware security keys, GitHub still requires the use of TOTP codes or SMS text messages. This can indeed feel limiting, especially for users who are well-versed in account security management.

### Possible Actions and Recommendations

1. **Provide Feedback to GitHub**: Many platforms evolve their security policies based on user feedback. You can submit your concerns and suggestions directly to GitHub:
   - Use GitHub's [feedback form](https://support.github.com/contact/feedback?category=features) to detail your request.
   - Engage with GitHub support on social media platforms like Twitter, where they might have a support handle.

2. **Workarounds**:
   - **Minimize Exposure**: If you're forced to keep a TOTP or SMS method enabled, you can use a TOTP app that runs in a highly secure environment, minimizing the risk associated with this method.
   - **Secure the SMS Number**: If SMS is mandatory, ensure the phone number associated with your account is one that is not used publicly and is secured with the highest possible protections offered by your mobile carrier.

3. **Community Support**: 
   - Engage with the developer community on forums like GitHub Community Discussions or Stack Overflow. Sometimes, collective voices can push for changes more effectively than individual feedback.

4. **Stay Informed**: 
   - Keep an eye on GitHub’s [changelog](https://github.blog/changelog/) and blog for updates on security features. They may introduce more flexibility in their authentication policies in the future.

### Example Feedback

Here’s a template you might use to submit feedback to GitHub:

---

**Subject**: Request to Allow Removal of Legacy Authentication Methods

**Message**:

Dear GitHub Team,

I am writing to express my concern over the current requirement to keep either TOTP codes or SMS text messages enabled as authentication methods on my GitHub account. As a user who prioritizes strong security practices, I rely primarily on passkeys and hardware security keys for authentication.

Given that my main security key is always with me and I have additional backups securely stored, I do not require the use of what I consider to be less secure methods like TOTP or SMS. I manage my account security meticulously, similar to my enrollment in Google's Advanced Protection Program, and I believe many GitHub users share this level of security awareness.

I kindly request that GitHub consider allowing the removal of these legacy authentication methods to provide a more secure and flexible authentication experience for users like myself. 

Thank you for your consideration.

Best regards,

[Your Name]

---

**Reply by queenofcorgis** [reply]
@Sivamani-18 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
